Compass/AccountSettingsViewController.swift

```diff
@@ -0,0 +1,246 @@
+        imagePicker.allowsEditing = false
+        self.present(imagePicker, animated: true, completion: nil)
+    }
+    
+    func imagePickerController(_ picker: UIImagePickerController, didFinishPickingMediaWithInfo info: [String : Any]) {
+        if let image = info[UIImagePickerControllerOriginalImage] as? UIImage {
+            let compressedImage = UIImageJPEGRepresentation(image, 0.2)
+            self.avatarImage.image = UIImage(data:compressedImage!)
+        }
+        self.dismiss(animated: true, completion: nil)
+    }
+    
+    func createAlert(title: String, message: String) {
+        
+        let alert = UIAlertController(title: title, message: message, preferredStyle: UIAlertControllerStyle.alert)
+        alert.addAction(UIAlertAction.init(title: "OK", style: .default, handler: { (action) in
+            self.dismiss(animated: true, completion: nil)
+        }))
+        
+        self.present(alert, animated: true, completion: nil)
+    }
+    
+    @objc func saveUserDetails(){
+
+        self.activityIndicator = UIActivityIndicatorView(frame: CGRect.zero)
+        self.activityIndicator.center = self.view.center
+        self.activityIndicator.hidesWhenStopped = true
+        self.activityIndicator.activityIndicatorViewStyle = UIActivityIndicatorViewStyle.gray
+        self.view.addSubview(self.activityIndicator)
+        self.activityIndicator.startAnimating()
+        UIApplication.shared.beginIgnoringInteractionEvents()
+        
+        userData.name = self.nameTextField.text
+        userData.userInfo = self.userInforTextField.text
+        userData.avatar = self.avatarImage.image!
+        
+        par.saveUserData(userData: userData) { (success, error) in
+            self.activityIndicator.stopAnimating()
+            UIApplication.shared.endIgnoringInteractionEvents()
+            
+            guard error == nil else {
+                var displayErrorMessage = "Please try again later"
+                let error = error as NSError?
+                if let errorMessage = error?.userInfo["error"] as? String {
+                    displayErrorMessage = errorMessage
+                }
+                self.createAlert(title: "Error:", message: displayErrorMessage)
+                return
+            }
+            
+            if success {
+                //user Location saved as a seperate object, this also need to be updated when username changed
+                if let id = UserDefaults.standard.string(forKey: "locationObjectId") {
+                    self.par.updateUserLocation(classN: "Locations", id: id, location: nil)
+                }
+                self.navigationController?.popViewController(animated: true)
+            }
+        }
+    }
+    
+    override func touchesBegan(_ touches: Set<UITouch>, with event: UIEvent?) {
+        nameTextField.resignFirstResponder()
+        userInforTextField.resignFirstResponder()
+    }
+    
+}
```

Comment:
Besides the few minor comments, this whole class is looking really good 👍

Reply:
That is great, thanks :)